hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -703,6 +710,78 @@ public String toString() {
         + "}";
   }
 
+  /**
+   * Gets the ACL of a key.
+   *
+   * @param path Path to get
+   * @return AclStatus describing the ACL of the key
+   * @throws IOException if an ACL could not be read
+   * @throws UnsupportedOperationException if the operation is unsupported
+   *         (default outcome).
+   */
+  @Override
+  public AclStatus getAclStatus(Path path) throws IOException {
+    Map<String, List<OzoneAcl>> ozoneAcls = adapter.getAcl(pathToKey(path));
+    String userAcls =
+        JsonUtils.toJsonStringWithDefaultPrettyPrinter(
+            ozoneAcls.get(ACLIdentityType.USER.toString()));
+    String groupAcls =
+        JsonUtils.toJsonStringWithDefaultPrettyPrinter(
+            ozoneAcls.get(ACLIdentityType.GROUP.toString()));
+    return new AclStatus.Builder()
+            .owner(userAcls)
+            .group(groupAcls)
+            .build();
+  }
+
+  /**
+   * Fully replaces ACL of a key, discarding all existing entries.
+   *
+   * @param path Path to modify
+   * @param aclSpec List describing modifications, which must include entries
+   *   for user, group for compatibility with permission bits.
+   * @throws IOException if an ACL could not be modified
+   * @throws UnsupportedOperationException if the operation is unsupported
+   *         (default outcome).
+   */
+  @Override
+  public void setAcl(Path path, List<AclEntry> aclSpec) throws IOException {
+    adapter.setAcl(pathToKey(path), aclSpecToString(aclSpec));
+  }
+
+  public String toAclString(AclEntry aclEntry) {
```

Comment:
I think we are converting AclEntry to form which can be parsed by OzoneAcl. Would it make sense to move this function to OzoneAcl class?